lib/serum/build/file_processor.ex

```diff
@@ -163,6 +167,10 @@ defmodule Serum.Build.FileProcessor do
 
   @doc false
   @spec generate_lists([map()], Project.t()) :: Result.t({[PostList.t()], tag_counts()})
+  def generate_lists(compact_posts, proj)
+
+  def generate_lists([], _proj), do: {:ok, []}
```

Comment:
Oh I nearly missed this one. According to the function spec, this function must return a 2-element tuple of lists on success.

```suggestion
  def generate_lists([], _proj), do: {:ok, {[], []}}
```

You may amend your commit and force-push to fix this. My bad!